src/HotChocolate/Core/src/Subscriptions.Postgres/PostgresMessageEnvelope.cs

```diff
@@ -12,20 +14,20 @@ namespace HotChocolate.Subscriptions.Postgres;
     private const byte separator = (byte)':';
     private const byte _messageIdLength = 24;
 
-    public PostgresMessageEnvelope(string topic, string payload)
+    public PostgresMessageEnvelope(string topic, string payload, int maxMessagePayloadSize)
     {
         Topic = topic;
-        Payload = payload;
+        FormattedPayload = Format(topic, payload, maxMessagePayloadSize);
     }
 
     public string Topic { get; }
 
-    public string Payload { get; }
+    public string FormattedPayload { get; }
 
-    public string Format()
+    private static string Format(string topic, string payload, int maxMessagePayloadSize)
```

Comment:
I personally would prefer a public static factory method that does the formatting and then passes `topic` and `payload` to the envelope ctor.

Reply:
yeah, this is fair, i wasn't sure about doing that extra work in the constructor, a factory would be better. Ill make that change